ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/subscription/blockheaders/NewBlockHeadersSubscriptionServiceTest.java

```diff
@@ -251,11 +232,9 @@ private void simulateAddingBlockOnCanonicalChain(final Block block) {
     verify(blockchainQueriesSpy, times(1)).getBlockchain();
   }
 
-  private void simulateAddingBlockOnNonCanonicalChain() {
-    final BlockBody blockBody = new BlockBody(Collections.emptyList(), Collections.emptyList());
-    final Block testBlock = new Block(blockHeader, blockBody);
+  private void simulateAddingBlockOnNonCanonicalChain(final Block block) {
     newBlockHeadersSubscriptionService.onBlockAdded(
-        BlockAddedEvent.createForFork(testBlock), blockchainQueriesSpy.getBlockchain());
+        BlockAddedEvent.createForFork(block), blockchainQueriesSpy.getBlockchain());
```

Comment:
Now the simulate method forces the call of `onBlockAdded` but since you have the blockchain, these should automatically be called when you subscribe to NewHeaders.

Basically, the tests could look like:

* Create blockchain
* Subscribe to BlockHeaders
* Add blocks
* Check what queries were made

All without ever calling `onBlockAdded` manually.